images_to_samples.py

```diff
@@ -183,54 +187,65 @@ def main(params):
-        if params['sample']['mask_reference']:
-            np_label_raster = mask_image(np_input_image, np_label_raster)
-
-        if info['dataset'] == 'trn':
-            out_file = trn_hdf5
-        elif info['dataset'] == 'val':
-            out_file = val_hdf5
-        elif info['dataset'] == 'tst':
-            out_file = tst_hdf5
-        else:
-            raise ValueError(f"Dataset value must be trn or val or tst. Provided value is {info['dataset']}")
-
-        np_label_raster = np.reshape(np_label_raster, (np_label_raster.shape[0], np_label_raster.shape[1], 1))
-        number_samples, number_classes = samples_preparation(np_input_image,
-                                                             np_label_raster,
-                                                             params['global']['samples_size'],
-                                                             params['sample']['samples_dist'],
-                                                             number_samples,
-                                                             number_classes,
-                                                             out_file,
-                                                             info['dataset'],
-                                                             params['sample']['min_annotated_percent'])
-
-        print(info['tif'])
-        print(number_samples)
-        out_file.flush()
+    with tqdm(list_data_prep) as _tqdm:
+        for info in _tqdm:
+
+            if bucket_name:
+                bucket.download_file(info['tif'], "Images/" + info['tif'].split('/')[-1])
+                info['tif'] = "Images/" + info['tif'].split('/')[-1]
+                if info['gpkg'] not in gpkg_file:
+                    gpkg_file.append(info['gpkg'])
+                    bucket.download_file(info['gpkg'], info['gpkg'].split('/')[-1])
+                info['gpkg'] = info['gpkg'].split('/')[-1]
+
+            assert_band_number(info['tif'], params['global']['number_of_bands'])
+
+            _tqdm.set_postfix(OrderedDict(file=f'{info["tif"]}', sample_size=params['global']['samples_size']))
+
+            # Read the input raster image
+            np_input_image = image_reader_as_array(info['tif'])
+
+            # Validate the number of class in the vector file
+            validate_num_classes(info['gpkg'], params['global']['num_classes'], info['attribute_name'])
+
+            # Burn vector file in a raster file
+            np_label_raster = vector_to_raster(info['gpkg'], info['tif'], info['attribute_name'])
+
+            # Guidelines for pre-processing: http://cs231n.github.io/neural-networks-2/#datapre
+            # Scale arrays to values [0,1]. Default: will scale. Useful if dealing with 8 bit *and* 16 bit images.
+            scale = params['sample']['scale_data'] if params['sample']['scale_data'] else True
+            if scale:
+                min, max = params['sample']['scale_data']
```

Comment:
min and max also refers to python's built-in functions. Rename for clarification.